website/src/docs/hotchocolate/v14/integrations/entity-framework.md

```diff
@@ -2,204 +2,176 @@
+        Guid id,
+        ApplicationDbContext dbContext) // ⬅️
+    {
+        return await dbContext.Books.FindAsync(id);
+    }
+}
+```
 
-builder.Services
-    .AddGraphQLServer()
-    .RegisterDbContext<ApplicationDbContext>(DbContextKind.Pooled)
-    .AddQueryType<Query>();
+</Implementation>
+<Code>
 
-public class Query
+```csharp
+public sealed class QueryType : ObjectType
 {
-    public Foo GetFoo(ApplicationDbContext dbContext)
-        => // Omitted code for brevity
+    protected override void Configure(
+        IObjectTypeDescriptor descriptor)
+    {
+        descriptor.Name(OperationTypeNames.Query);
+
+        descriptor
+            .Field("bookById")
+            .Argument("id", a => a.Type<NonNullType<UuidType>>())
+            .Resolve(async ctx => await ctx
+                .Service<IDbContextFactory<ApplicationDbContext>>()
+                // ⬆️
+                .CreateDbContext()
+                .Books
+                .FindAsync(ctx.ArgumentValue<Guid>("id")));
+    }
 }
 ```
 
-When injecting a `DbContext` using the `DbContextKind.Pool`, Hot Chocolate will retrieve one `DbContext` instance from the pool for each invocation of a resolver. Once the resolver has finished executing, the instance will be returned to the pool.
+</Code>
+<Schema>
 
-Since each resolver invocation is therefore working with a "transient" `DbContext` instance, Hot Chocolate can parallelize the execution of resolvers using this `DbContext`.
+Take a look at the implementation-first or code-first example.
 
-# Working with a pooled DbContext
+</Schema>
+</ExampleTabs>
 
-If you have registered your `DbContext` using [DbContextKind.Pooled](#dbcontextkindpooled) you are on your way to squeeze the most performance out of your GraphQL server, but unfortunately it also changes how you have to use the `DbContext`.
+> Warning: As shown above, you still need to add your `DbContextFactory` to the dependency injection container, by calling `AddDbContextFactory<T>` or `AddPooledDbContextFactory<T>`. `RegisterDbContextFactory<T>` on its own is not enough.
 
-For example you need to move all of the configuration from the `OnConfiguring` method inside your `DbContext` into the configuration action on the `AddPooledDbContextFactory` call.
+# Working with a DbContext factory
 
-You also need to access your `DbContext` differently. In the following chapters we will take a look at some of the changes you have to make.
+When you use a DbContext factory, you need to access your DbContext differently if it is not being directly injected into a resolver. In the following sections we will take a look at some of the changes that you need to make.
 
 ## DataLoaders
 
-When creating DataLoaders that need access to your `DbContext`, you need to inject the `IDbContextFactory<T>` using the constructor.
+When creating DataLoaders that need access to your DbContext, you need to inject the `IDbContextFactory<T>` using the constructor.
```

Comment:
This is example is with the older DataLoader implementation, not the GreenDonut source-generator implementation, correct?

Reply:
Yes.